### ISSUE TITLE
`no-unpublished-import`/`require` error importing devDependencies

### ISSUE THREAD

**Issue by latipun7** [mysticatea/eslint-plugin-node#225]
I have private project:

`package.json` (no `files` field and no `.npmignore`)

```json
{
  "name": "private-project",
  "private": true,
  "dependencies": {
    "axios": "*"
  },
  "devDependencies": {
    "@babel/core": "*"
  }
```

When I load devDependencies, eslint throw an error:

`index.js`

```js
import axios from 'axios';
import babel from '@babel/core'; // this throw an error: "@babel/core" is not published. eslint(node/no-unpublished-import)

// ...
```

with this following eslint configurations:

`.eslintrc.js`

```js
module.export = {
  settings: {
    node: {
      allowModules: [],
      resolvePaths: [],
      tryExtensions: ['.js', '.json', '.node'],
    }
  },
  rules: {
    'node/no-unpublished-import': 'error',
    'node/no-unpublished-require': 'error',
  }
}
```

But if I move babel in `dependencies`, it does not throw an error

`package.json`

```json
{
  "name": "private-project",
  "private": true,
  "dependencies": {
    "@babel/core": "*",
    "axios": "*"
  }
```
`index.js`

```js
import axios from 'axios';
import babel from '@babel/core'; // this **does not** throw an error

// ...
```

Do I configure something wrong or is that expected behavior? Thanks.

**More info**:
installed eslint version: 6.8.0
installed eslint-plugin-node version: 11.1.0

**Comment by mysticatea** [comment]
That's the purpose of `no-unpublished-import`/`require` rule. We cannot import the `devDependencies`  after publishing the package.

**Comment by latipun7** [comment]
I see, thank you.

So does the feature for issue #77 would be good? With `'no-unpublished-import': 'error'`, rule translated as `'no-unpublished-import': 'off'` when in `package.json` has `"private": true` in the field. If no `private` filed present at `package.json`, the rule applied as is.

**Comment by ronkorving** [comment]
It would be nice for eslintrc's for test-folders and perhaps CLI-script-folders if we could configure this rule to also consider dev-dependencies. Is that something worth considering, @mysticatea ?

**Comment by mysticatea** [comment]
@latipun7 at least, I use `no-unpublished-import` rule for private packages. Because I use `npm pack` and `npm install` to install my private packages.

@ronkorving it's valid if non-published files use dev-dependencies. You can specify non-published files by `.npmignore` or the `files` property in `package.json`.

**Comment by ronkorving** [comment]
@mysticatea 
Oh wow! So for a unpublished project, like an application, just an .npmignore file like:

```
*
```

would be enough?

**Comment by latipun7** [comment]
> Oh wow! So for a unpublished project, like an application, just an .npmignore file like:
> 
> ```
> *
> ```
> 
> would be enough?

Yes, it is.

Thank you @mysticatea .

**Comment by ronkorving** [comment]
Awesome, thanks for clarifying, @latipun7 !
